### PR TITLE
#105 Implement Pull To Refresh integration

### LINKED ISSUE #105: [Integrate] As a user, when I pull to refresh, the application is updated with the latest data
## Why
Add integration when Pull to Refresh - calling to fetch the latest data from the API (must fetch from network not pulling from cache).

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #105 

## What happened 👀

Integrate the pull to refresh functionality.
 
## Insight 📝

- Add the capability to force a network request (still caching them for later usage).
- Integrate the behavior on the UI.
 
## Proof Of Work 📹

Notice the data source is coming correctly from .network:

https://user-images.githubusercontent.com/13435717/128812609-11e11dd9-d99b-4d7f-8973-387c36f1f427.mov

### CODE REVIEW

File: lib/pages/home/home_controller.dart [hunk 1]
```diff
@@ -67,14 +67,15 @@ class HomeController extends GetxController {
     final getSurveysUseCase = Get.find<GetSurveysUseCase>();
     // ignore: invalid_use_of_protected_member
     if (_surveys.value.isEmpty) {
-      final surveys = await getSurveysUseCase.call("");
+      final surveys = await getSurveysUseCase.call(GetSurveysInput(cursor: ""));
```
Comment (markgravity) [inline lib/pages/home/home_controller.dart:70]: What does `cursor: ""` mean? 🤔

Comment (sleepylee) [reply lib/pages/home/home_controller.dart:70]: start from empty 😅  `cursor` in our `graphql API` is the key id of the item you wanted to point to

image

Comment (sleepylee) [reply lib/pages/home/home_controller.dart:70]: image

here is an example of how I formed the `query`. In our graphql API, pagination doesn't count by params in the request like `page=0`, `page=1&count=10` like in the REST API; instead, you can specify a `cursor` and query the `amount` of result `after` that cursor, if the `amount` is static, then you can always iterate through the last content (pagination).

You can read more at [the official GraphQL documentation]([https://graphql.org/learn/pagination/#complete-connection-model) for more examples

File: lib/pages/home/home_page.dart [hunk 2]
```diff
@@ -10,23 +10,25 @@ import 'package:survey/pages/home/survey_carousel/survey_carousel_card.dart';
 class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    RefreshController _refreshController =
+    RefreshController pullToRefreshController =
         RefreshController(initialRefresh: false);
+
+    PageController pagingController = PageController();
```
Comment (manh-t) [inline lib/pages/home/home_page.dart:16]: I'm afraid that we still need `dispose` for this `PageController` 🤕 https://api.flutter.dev/flutter/widgets/PageController-class.html

Comment (sleepylee) [reply lib/pages/home/home_page.dart:16]: @manh-t this page is a StatelessWidget and it doesn't have `dispose()` 😅  as the widget doesn't mean to have a new state once it's created so disposing the state isn't required imo

Comment (manh-t) [reply lib/pages/home/home_page.dart:16]: @sleepylee `as the widget doesn't mean to have a new state once it's created so disposing the state isn't required`, I agree with you somehow on this point. 🤸‍♂️

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**